gquant/dataframe_flow/taskGraph.py

```diff
@@ -388,20 +523,21 @@ def process(self, inputs):
             i.flow()
 
         results_dfs_dict = outputs_collector_node.input_df
+        Result = namedtuple('Result', " ".join(outputs).replace('.', '_'))
```

Comment:
It's a bit awkward to rename the outputs (inconsistent with notation we're using `<node_id>.<out_port>`). I was testing various ways to achieve the tuple unpacking API, but be able to have a dict like container. Maybe just overriding a dict (or OrderedDict):
```
class Results(OrderedDict): 
    def __iter__(self): 
        return iter(self.values())

mdict = Results([('x.y', 5), ['u.v', 6]])

print(mdict)
Results([('x.y', 5), ('u.v', 6)])

In [61]: x, y = mdict                                                                                                                                                                                              

In [64]: x, y                                                                                                                                                                                                      
Out[64]: (5, 6)

n [65]: def tfn(tvar): 
    ...:     return tvar 
    ...:                                                                                                                                                                                                           

In [66]: (x,y) = tfn(mdict)                                                                                                                                                                                          

In [67]: x, y                                                                                                                                                                                                      
Out[67]: (5, 6)

```

Above behaves like a dict, but unpacks the values instead of keys. That will make it compatible with tuple API in existing notebooks. Otherwise it can be used like a dictionary.

Reply:
I created a Result object to have all the desirable behaviors we want
```python
class Results(object):

    def __init__(self, values):
        self.values = tuple([i[1] for i in values])
        self.__dict = OrderedDict(values)

    def __iter__(self):
        return iter(self.values)

    def __getitem__(self, key):
        if isinstance(key, int):
            return self.values[key]
        else:
            return self.__dict[key]

    def __len__(self):
        return len(self.values)

    def __repr__(self):
        return "Results"+self.__dict.__repr__()[11:]

    def __str__(self):
        return "Results"+self.__dict.__str__()[11:]

    def __contains__(self, key):
        return True if key in self.__dict else False
```
you can do
```python
x, y = mdict 
mydict[0]
mydict[1]
mydict['x.y']
mydict['u.v']
```

Reply:
Looks good.